### PR TITLE
Fixing compare operators of Git::Reference

### PR DESCRIPTION
**Pull request by antis81**
The first implementation of object comparison of two `Git::Reference` objects is wrong and error prone (sorry for that). Especially as we are working with smart pointers, we need to worry about when an object is marked equal. This PR adds a simple fix for comparing references of Git::Reference objects (sometimes I hate language :smile:). Question here: Should we compare the `d` member here?

To compare the destination (symbol or OID) of references, the new method `compare` should be used. This wraps exactly `git_reference_cmp` from lg2.

Note: This enforces a correction to the behaviour of the MGV module `RefsViews`, currently relying on the == operator.

### CODE REVIEW

File: libGitWrap/Reference.cpp [hunk 1]
```diff
@@ -69,10 +69,7 @@
 
     bool Reference::operator==(const Reference &other) const
     {
-        if( d && other.d )
-            return git_reference_cmp( d->mRef, other.d->mRef ) == 0;
-
-        return false;
+        return this == &other;
```
Comment (scunz) [inline libGitWrap/Reference.cpp]: > Question here: Should we compare the d member here?

Indeed, it should be _just_: `return d != other.d;`. `this` and `&other` can be different, even if their actual data pointers are the same. i.e. your implementation would deem this code to return `false`:

```
bool foo( Reference& r ) {
    Reference r2 = r;
    return r == r2;
}
```

**Comment by scunz** [comment]
Basically I am fine with this change, but I'm not sure at all that I understand the motivation behind this change.

> To compare the destination (symbol or OID) of references, the new method compare should be used. This wraps exactly git_reference_cmp from lg2.

Does this mean, that `git_reference_cmp` compares two different refs (like `refs/heads/a` and `refs/heads/b`) as equal, if both are pointing either to the same symbol or the same SHA!?

**Comment by antis81** [comment]
> Does this mean, that git_reference_cmp compares two different refs (like refs/heads/a and refs/heads/b) as equal, if both are pointing either to the same symbol or the same SHA!?

Yeah, exactly that. The result codes from lg2 have the following meanings:
- -1 -> ref types differ, first ref is symbolic
- 1 -> either ref types differ, second is symbolic ( or the position of first difference in oid or string )
- >=1 -> position of first difference in ref names or oid's
- 0 -> matching:
  - For symbolic refs, the target names match
  - For direct refs, the oid's match

This is used in the RefsViews to show branches pointing to HEAD marked "lighter orange" and the active branch in "orange". This is actually very useful.

**Comment by scunz** [comment]
Then let's use this code :-)

**Pull request merged**